### PR TITLE
adding response_hook to elastic instrumentation

### PR DESCRIPTION
**Pull request by ItayGibel-heliosphere**
# Description

`opentelemetry-instrumentation-elasticsearch`: added response_hook callback passed as an argument to the instrument method, enabling the users to add data from the response to the span attributes

## Type of change

- [X] New feature (non-breaking change which adds functionality)
- [X] This change requires a documentation update

# How Has This Been Tested?

- [X] `test_response_hook` -  a new unit test testing the hook activation

# Does This PR Require a Core Repo Change?

- [ ] Yes. - Link to PR: 
- [X] No.

# Checklist:

See [contributing.md](https://github.com/open-telemetry/opentelemetry-python-contrib/blob/main/CONTRIBUTING.md) for styleguide, changelog guidelines, and more.

- [X] Followed the style guidelines of this project
- [X] Changelogs have been updated
- [X] Unit tests have been added
- [ ] Documentation has been updated

### CODE REVIEW

File: instrumentation/opentelemetry-instrumentation-elasticsearch/src/opentelemetry/instrumentation/elasticsearch/__init__.py [hunk 1]
```diff
@@ -143,13 +146,16 @@ def wrapper(wrapped, _, args, kwargs):
...
     return wrapper
+
+
+def _default_response_hook(span, response):
+    if isinstance(response, dict) and span.is_recording():
+        for member in _ATTRIBUTES_FROM_RESULT:
+            if member in response:
+                span.set_attribute(
+                    "elasticsearch.{0}".format(member), str(response[member]),
+                )
```
Comment (owais) [inline instrumentation/opentelemetry-instrumentation-elasticsearch/src/opentelemetry/instrumentation/elasticsearch/__init__.py:161]: Any reason to move logic into a "default hook"? Hooks give users the ability to perform additional operations on or with the span data. There shouldn't be any default hooks containing any logic that gets replaced by a user provided hook. Please move this back to the previous place.

**Comment by owais** [comment]
This looks fine but it's a bit strange to just have a response hook without request.

**Review by owais — APPROVED** [review]

File: instrumentation/opentelemetry-instrumentation-elasticsearch/src/opentelemetry/instrumentation/elasticsearch/__init__.py [hunk 2]
```diff
@@ -97,17 +97,23 @@ def _instrument(self, **kwargs):
         """
         tracer_provider = kwargs.get("tracer_provider")
```
Comment (lzchen) [inline instrumentation/opentelemetry-instrumentation-elasticsearch/src/opentelemetry/instrumentation/elasticsearch/__init__.py:98]: Can you update the usage docs with an example of using the request/response hooks with their function signatures?

Comment (ItayGibel-heliosphere) [reply instrumentation/opentelemetry-instrumentation-elasticsearch/src/opentelemetry/instrumentation/elasticsearch/__init__.py:133]: done.

Comment (ItayGibel-heliosphere) [reply instrumentation/opentelemetry-instrumentation-elasticsearch/src/opentelemetry/instrumentation/elasticsearch/__init__.py:133]: Done.

**Review by lzchen — APPROVED** [review]
@ItayGibel-heliosphere 
Updates from others aren't allowed in this PR. Can you rebase with main so we can merge this in?

**Pull request merged**